File: planning-idsdp-rms-frontend/src/app/main/modules/rpm/models/ResearcherProposalUploadDoc.ts
export class ResearcherProposalUploadDoc {

    id: number;
    uuid: string;
    researcherProposalId: number;
    researcherProposalUuid: string;
    stDocumentTypeId: number;
    docName: string;
    briefOnDocument: string;
    fileDownloadUrl: string;
    bucketName: string;
    fileName: string;
    isEditable: number;
    deleted: number = 0;
    documentTypeName : string
    // createdOn: Date;

}

File: planning-idsdp-rms-rpm-backend/src/main/java/com/ibcs/idsdp/rpm/model/domain/ResearcherProposalUploadDoc.java
package com.ibcs.idsdp.rpm.model.domain;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

import org.hibernate.annotations.Cache;
import org.hibernate.annotations.CacheConcurrencyStrategy;
import org.hibernate.validator.constraints.Length;

import com.ibcs.idsdp.common.model.domain.BaseEntity;

import lombok.Data;

@Table(name = "m1_researcher_proposal_upload_doc")
@Entity
@Data
@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
public class ResearcherProposalUploadDoc extends BaseEntity {

	@NotNull
	@ManyToOne
	@JoinColumn(name = "researcher_proposal_id")
	private ResearcherProposal researcherProposal;

	private Long stDocumentTypeId;

	@NotNull
	private String docName;


	@Column(columnDefinition = "TEXT")
	private String briefOnDocument;


	@Column(length = 5000)
	private String fileDownloadUrl;

	@Column(length = 5000)
	private String bucketName;

	@Column(length = 5000)
	private String fileName;

	private Integer isEditable;

}
